Module1/Dockerfile

```diff
@@ -0,0 +1,62 @@
+# syntax=docker/dockerfile:1
+
+# Comments are provided throughout this file to help you get started.
+# If you need more help, visit the Dockerfile reference guide at
+# https://docs.docker.com/engine/reference/builder/
+
+ARG PYTHON_VERSION=3.11.4
+FROM python:${PYTHON_VERSION}-slim as base
+
+FROM base as development
+
+# Prevents Python from writing pyc files.
+ENV PYTHONDONTWRITEBYTECODE=1
+
+# Keeps Python from buffering stdout and stderr to avoid situations where
+# the application crashes without emitting any logs due to buffering.
+ENV PYTHONUNBUFFERED=1
+
+WORKDIR /app
+# Copy the source code into the container.
+COPY . /app
+
+RUN apt-get update && apt-get install -y curl
+RUN curl -sSL https://install.python-poetry.org | python3 -
+ENV PATH="/root/.local/bin:/app/.local/bin:$PATH"
+
+# RUN poetry install --no-interaction --no-ansi
+RUN poetry update && poetry install
```

Comment:
Bonus comment, I'd typically avoid using a `poetry update` in your Dockerfile - that means the `poetry.lock` (and therefore your dependencies) will potentially be changing each time you build the image which might give you inconsistent behaviour. We do want to run the update regularly, but we want those changes to be checked in & committed to our codebase for consistency each time (as it is, we aren't gaining very much from having a `poetry.lock` here)